src/main.cpp

```diff
@@ -174,19 +138,27 @@ int Main(vector<string> args)
 		
 		parser[r.c_str()].enter = [r](const char* s, size_t n, any& dt) {
 			auto& indent = *dt.get<int*>();
-			for(int i=0; i<indent; i++){cout << "|  ";}
-			string beginning(s, n);
-			if(beginning.length()>10){beginning = beginning.substr(0,10) + "...";}
-			for(auto& c:beginning){if(c == '\n'){c = '\\';}}
-			cout << r << " => \"" << beginning << "\"?" << endl;
+			cout << repeat("|  ", indent) << r << " => \"" << shorten(string(s, n), 40) << "\"?" << endl;
```

Comment:
nit: `shorten(string(s, std::min(38, n)), 40)` may avoid an unnecessary copy of a long string.